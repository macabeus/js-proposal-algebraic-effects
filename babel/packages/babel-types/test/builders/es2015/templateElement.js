import * as t from "../../..";

describe("builders", function() {
  describe("es2015", function() {
    describe("templateElement", function() {
      it("should validate", function() {
        expect(
          t.templateElement({ raw: "foo", cooked: "foo" }),
        ).toMatchSnapshot();

        expect(t.templateElement({ raw: "foo" })).toMatchSnapshot();

        expect(() =>
          t.templateElement({ raw: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateElement({ raw: "foo", cooked: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() => t.templateElement("foo")).toThrowErrorMatchingSnapshot();
      });
    });
    describe("templateLiteral", function() {
      it("should validate", function() {
        const foo = t.templateElement({ raw: "foo" });
        const bar = t.templateElement({ raw: "bar" });

        const baz = t.stringLiteral("baz");
        const qux = t.stringLiteral("qux");

        expect(t.templateLiteral([foo], [])).toMatchSnapshot();

        expect(t.templateLiteral([foo, bar], [baz])).toMatchSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar], [baz, qux]),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar], []),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral({}, [baz]),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar]),
        ).toThrowErrorMatchingSnapshot();
      });
    });
  });
});
