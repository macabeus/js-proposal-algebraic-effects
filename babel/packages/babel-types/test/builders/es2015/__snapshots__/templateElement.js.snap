// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders es2015 templateElement should validate 1`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "foo",
    "raw": "foo",
  },
}
`;

exports[`builders es2015 templateElement should validate 2`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "raw": "foo",
  },
}
`;

exports[`builders es2015 templateElement should validate 3`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got number"
`;

exports[`builders es2015 templateElement should validate 4`] = `
"Property value of TemplateElement expected to have the following:
Property cooked expected type of string but got number"
`;

exports[`builders es2015 templateElement should validate 5`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got undefined"
`;

exports[`builders es2015 templateLiteral should validate 1`] = `
Object {
  "expressions": Array [],
  "quasis": Array [
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "raw": "foo",
      },
    },
  ],
  "type": "TemplateLiteral",
}
`;

exports[`builders es2015 templateLiteral should validate 2`] = `
Object {
  "expressions": Array [
    Object {
      "type": "StringLiteral",
      "value": "baz",
    },
  ],
  "quasis": Array [
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "raw": "foo",
      },
    },
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "raw": "bar",
      },
    },
  ],
  "type": "TemplateLiteral",
}
`;

exports[`builders es2015 templateLiteral should validate 3`] = `
"Number of TemplateLiteral quasis should be exactly one more than the number of expressions.
Expected 3 quasis but got 2"
`;

exports[`builders es2015 templateLiteral should validate 4`] = `
"Number of TemplateLiteral quasis should be exactly one more than the number of expressions.
Expected 1 quasis but got 2"
`;

exports[`builders es2015 templateLiteral should validate 5`] = `"Property quasis expected type of array but got object"`;

exports[`builders es2015 templateLiteral should validate 6`] = `"Property expressions expected type of array but got null"`;
